homeassistant/components/accuweather/__init__.py

```diff
@@ -73,12 +76,25 @@ def __init__(
         api_key: str,
         location_key: str,
         forecast: bool,
+        name: str,
     ) -> None:
         """Initialize."""
         self.location_key = location_key
         self.forecast = forecast
         self.is_metric = hass.config.units.is_metric
-        self.accuweather = AccuWeather(api_key, session, location_key=self.location_key)
+        self.accuweather = AccuWeather(api_key, session, location_key=location_key)
+        self.device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, location_key)},
+            manufacturer=MANUFACTURER,
+            name=name,
+            # You don't need to provide specific details for the URL,
+            # so passing in _ characters is fine if the location key
+            # is correct
+            configuration_url="http://accuweather.com/en/"
```

Comment:
Can we use a parenthesis and indent the string parts so it's easier to read this?

Reply:
Done